### PR TITLE
Operator Api | Public Transport Alternatives

### PR DESCRIPTION
**Pull request by tom-ioki**
This adds all models that are necessary for public transport alternatives.

### CODE REVIEW

File: lib/ioki/model/operator/ride.rb [hunk 1]
```diff
@@ -28,6 +28,11 @@ class Ride < Base
                   on:   :read,
                   type: :boolean
 
+        attribute :booking_solution,
```
Comment (phylor) [inline lib/ioki/model/operator/ride.rb:31]:
```suggestion
        attribute :booked_solution,
```

Right? 🙈

**Review by phylor — COMMENTED** [review]
Haven't checked all of it yet, just a quick review so far :slightly_smiling_face:

File: lib/ioki/model/operator/ride.rb (hunk 1 above)
Comment (tom-ioki) [reply lib/ioki/model/operator/ride.rb:31]: yes 😓

File: lib/ioki/model/operator/public_transport_stop.rb [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+        attribute :country,
+                  type: :string,
+                  on:   :read
+
+        attribute :county,
+                  type: :string,
+                  on:   :read
+
+        attribute :current_arrival,
+                  type: :string,
```
Comment (phylor) [inline lib/ioki/model/operator/public_transport_stop.rb:24]:
```suggestion
                  type: :date_time,
```

Haven't tested it, but maybe we want to have a `DateTime` object here?

In the passenger API we defined it as `string` as well. Maybe it's more comfortable for you to have a `DateTime` object? Don't really have an opinion here, whatever is easier for you.

File: lib/ioki/model/operator/public_transport_stop.rb [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+                  on:   :read
+
+        attribute :current_departure,
+                  type: :string,
```
Comment (phylor) [inline lib/ioki/model/operator/public_transport_stop.rb:28]:
```suggestion
                  type: :date_time,
```

Same as for `current_arrival`

File: lib/ioki/model/operator/public_transport_stop.rb [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+        attribute :current_platform,
+                  type: :string,
+                  on:   :read
+
+        attribute :dhid,
+                  type: :string,
+                  on:   :read
+
+        attribute :display_times,
```
Comment (phylor) [inline lib/ioki/model/operator/public_transport_stop.rb:39]: These could also be `DateTime` objects if you want them to be.

File: lib/ioki/model/operator/public_transport_stop.rb [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+        attribute :location_name,
+                  type: :array,
+                  on:   :read
+
+        attribute :postal_code,
+                  type: :array,
+                  on:   :read
+
+        attribute :scheduled_arrival,
+                  type: :string,
```
Comment (phylor) [inline lib/ioki/model/operator/public_transport_stop.rb:64]:
```suggestion
                  type: :date_time,
```

Depending on your decision above.

File: lib/ioki/model/operator/public_transport_stop.rb [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+                  on:   :read
+
+        attribute :scheduled_departure,
+                  type: :string,
```
Comment (phylor) [inline lib/ioki/model/operator/public_transport_stop.rb:68]:
```suggestion
                  type: :date_time,
```

Depending on your decision above.

**Review by phylor — APPROVED** [review]
Looks good! I'm not sure if we want the datetime strings to be parsed by ioki-ruby or not. We haven't done that in the passenger API, but maybe it's more comfortable to work with (we don't have to parse them again)?

File: lib/ioki/model/operator/public_transport_stop.rb (hunk 2 above)
Comment (tom-ioki) [reply lib/ioki/model/operator/public_transport_stop.rb:24]: Oh that's my bad. Of course these should be of type `date_time`. Thanks 🙇

File: lib/ioki/model/operator/public_transport_stop.rb (hunk 4 above)
Comment (tom-ioki) [reply lib/ioki/model/operator/public_transport_stop.rb:39]: type array is fine 👍

**Pull request merged**